Linker error when using the libmuonpi library

Functions ```muonpi::http::content_type::json()```and ```muonpi::http::content_type::html()``` are defined in the same header file as they are declared. This seems to lead to problems when using the library.
```
/usr/include/muonpi/http_response.h:72: multiple definition of `muonpi::http::content_type::html()'; CMakeFiles/rest_server.dir/rest_server/src/main.cpp.o:/usr/include/muonpi/http_response.h:72: first defined here
/usr/bin/ld: CMakeFiles/rest_server.dir/common_files/src/api.cpp.o: in function `muonpi::http::content_type::json()':
```

--- a/cmake/targets/http.cmake
+++ b/cmake/targets/http.cmake
@@ -6,6 +6,7 @@ if (LIBMUONPI_BUILD_HTTP) # libraries specific to the REST library
     set(HTTP_SOURCE_FILES
         "${PROJECT_SRC_DIR}/http_server.cpp"
         "${PROJECT_SRC_DIR}/http_request.cpp"
+        "${PROJECT_SRC_DIR}/http_response.cpp"
         "${PROJECT_SRC_DIR}/http_tools.cpp"
         )
 

--- a/include/muonpi/http_response.h
+++ b/include/muonpi/http_response.h
@@ -17,17 +17,17 @@ namespace muonpi::http {
  * @brief The content_type struct
  */
 struct content_type {
-  /**
-   * @brief html Get the content type string text/html
-   */
-  [[nodiscard]] static auto html() noexcept -> content_type;
+    /**
+     * @brief html Get the content type string text/html
+     */
+    [[nodiscard]] static auto html() noexcept -> content_type;
 
-  /**
-   * @brief json Get the content type string text/json
-   */
-  [[nodiscard]] static auto json() noexcept -> content_type;
+    /**
+     * @brief json Get the content type string text/json
+     */
+    [[nodiscard]] static auto json() noexcept -> content_type;
 
-  std::string string{};
+    std::string string {};
 };
 
 template <beast::http::status Status>
@@ -37,63 +37,67 @@ template <beast::http::status Status>
  */
 class LIBMUONPI_PUBLIC http_response {
 public:
-  /**
-   * @brief http_response
-   * @param req The type of request
-   * @param content The content type string
-   * @param application_name The name of the application in the http header
-   */
-  http_response(request_type &req, const content_type &content,
-                const std::string &application_name =
-                    "libmuonpi-" + Version::libmuonpi::string());
+    /**
+     * @brief http_response
+     * @param req The type of request
+     * @param content The content type string
+     * @param application_name The name of the application in the http header
+     */
+    http_response(request_type& req, const content_type& content,
+        const std::string& application_name = "libmuonpi-" + Version::libmuonpi::string());
 
-  /**
-   * @brief http_response
-   * Construct a default response
-   * @param req The request type
-   */
-  explicit http_response(request_type &req);
+    /**
+     * @brief http_response
+     * Construct a default response
+     * @param req The request type
+     */
+    explicit http_response(request_type& req);
 
-  /**
-   * @brief commit Complete the response
-   * @param body The body to attach to the response
-   */
-  [[nodiscard]] auto commit(std::string body) -> response_type;
+    /**
+     * @brief commit Complete the response
+     * @param body The body to attach to the response
+     */
+    [[nodiscard]] auto commit(std::string body) -> response_type;
 
-  /**
-   * @brief operator () Complete the response
-   * @param body The body to attach to the response
-   */
-  [[nodiscard]] auto operator()(std::string body) -> response_type;
+    /**
+     * @brief operator () Complete the response
+     * @param body The body to attach to the response
+     */
+    [[nodiscard]] auto operator()(std::string body) -> response_type;
 
 private:
-  response_type m_response;
+    response_type m_response;
 };
 
 template <beast::http::status Status>
-http_response<Status>::http_response(request_type &req,
-                                     const content_type &content,
-                                     const std::string &application_name)
-    : m_response{Status, req.version()} {
-  m_response.set(beast::http::field::server, application_name);
-  m_response.set(beast::http::field::content_type, content.string);
-  m_response.keep_alive(req.keep_alive());
+http_response<Status>::http_response(request_type& req,
+    const content_type& content,
+    const std::string& application_name)
+    : m_response { Status, req.version() }
+{
+    m_response.set(beast::http::field::server, application_name);
+    m_response.set(beast::http::field::content_type, content.string);
+    m_response.keep_alive(req.keep_alive());
 }
 
 template <beast::http::status Status>
-http_response<Status>::http_response(request_type &req)
-    : http_response<Status>{req, content_type::html()} {}
-
-template <beast::http::status Status>
-auto http_response<Status>::commit(std::string body) -> response_type {
-  m_response.body() = std::move(body);
-  m_response.prepare_payload();
-  return std::move(m_response);
+http_response<Status>::http_response(request_type& req)
+    : http_response<Status> { req, content_type::html() }
+{
 }
 
 template <beast::http::status Status>
-auto http_response<Status>::operator()(std::string body) -> response_type {
-  return std::move(commit(std::move(body)));
+auto http_response<Status>::commit(std::string body) -> response_type
+{
+    m_response.body() = std::move(body);
+    m_response.prepare_payload();
+    return std::move(m_response);
+}
+
+template <beast::http::status Status>
+auto http_response<Status>::operator()(std::string body) -> response_type
+{
+    return std::move(commit(std::move(body)));
 }
 
 } // namespace muonpi::http

--- a/include/muonpi/http_response.h
+++ b/include/muonpi/http_response.h
@@ -20,12 +20,12 @@ struct content_type {
     /**
      * @brief html Get the content type string text/html
      */
-    [[nodiscard]] inline static auto html() noexcept -> content_type;
+    [[nodiscard]] static auto html() noexcept -> content_type;
 
     /**
      * @brief json Get the content type string text/json
      */
-    [[nodiscard]] inline static auto json() noexcept -> content_type;
+    [[nodiscard]] static auto json() noexcept -> content_type;
 
     std::string string {};
 };
@@ -67,6 +67,35 @@ public:
 private:
     response_type m_response;
 };
+
+template <beast::http::status Status>
+http_response<Status>::http_response(request_type &req, const content_type &content, const std::string &application_name)
+    : m_response{Status, req.version()}
+{
+    m_response.set(beast::http::field::server, application_name);
+    m_response.set(beast::http::field::content_type, content.string);
+    m_response.keep_alive(req.keep_alive());
+}
+
+template <beast::http::status Status>
+http_response<Status>::http_response(request_type &req)
+    : http_response<Status>{req, content_type::html()}
+{
+}
+
+template <beast::http::status Status>
+auto http_response<Status>::commit(std::string body) -> response_type
+{
+    m_response.body() = std::move(body);
+    m_response.prepare_payload();
+    return std::move(m_response);
+}
+
+template <beast::http::status Status>
+auto http_response<Status>::operator()(std::string body) -> response_type
+{
+    return std::move(commit(std::move(body)));
+}
 
 } // namespace muonpi::http
 

--- a/include/muonpi/http_response.h
+++ b/include/muonpi/http_response.h
@@ -68,45 +68,6 @@ private:
     response_type m_response;
 };
 
-auto content_type::html() noexcept -> content_type
-{
-    return content_type { "text/html" };
-}
-
-auto content_type::json() noexcept -> content_type
-{
-    return content_type { "text/json" };
-}
-
-template <beast::http::status Status>
-http_response<Status>::http_response(request_type& req, const content_type& content, const std::string& application_name)
-    : m_response { Status, req.version() }
-{
-    m_response.set(beast::http::field::server, application_name);
-    m_response.set(beast::http::field::content_type, content.string);
-    m_response.keep_alive(req.keep_alive());
-}
-
-template <beast::http::status Status>
-http_response<Status>::http_response(request_type& req)
-    : http_response<Status> { req, content_type::html() }
-{
-}
-
-template <beast::http::status Status>
-auto http_response<Status>::commit(std::string body) -> response_type
-{
-    m_response.body() = std::move(body);
-    m_response.prepare_payload();
-    return std::move(m_response);
-}
-
-template <beast::http::status Status>
-auto http_response<Status>::operator()(std::string body) -> response_type
-{
-    return std::move(commit(std::move(body)));
-}
-
 } // namespace muonpi::http
 
 #endif // MUONPI_HTTP_RESPONSE_H

--- a/include/muonpi/http_response.h
+++ b/include/muonpi/http_response.h
@@ -17,17 +17,17 @@ namespace muonpi::http {
  * @brief The content_type struct
  */
 struct content_type {
-    /**
-     * @brief html Get the content type string text/html
-     */
-    [[nodiscard]] static auto html() noexcept -> content_type;
+  /**
+   * @brief html Get the content type string text/html
+   */
+  [[nodiscard]] static auto html() noexcept -> content_type;
 
-    /**
-     * @brief json Get the content type string text/json
-     */
-    [[nodiscard]] static auto json() noexcept -> content_type;
+  /**
+   * @brief json Get the content type string text/json
+   */
+  [[nodiscard]] static auto json() noexcept -> content_type;
 
-    std::string string {};
+  std::string string{};
 };
 
 template <beast::http::status Status>
@@ -37,64 +37,63 @@ template <beast::http::status Status>
  */
 class LIBMUONPI_PUBLIC http_response {
 public:
-    /**
-     * @brief http_response
-     * @param req The type of request
-     * @param content The content type string
-     * @param application_name The name of the application in the http header
-     */
-    http_response(request_type& req, const content_type& content, const std::string& application_name = "libmuonpi-" + Version::libmuonpi::string());
+  /**
+   * @brief http_response
+   * @param req The type of request
+   * @param content The content type string
+   * @param application_name The name of the application in the http header
+   */
+  http_response(request_type &req, const content_type &content,
+                const std::string &application_name =
+                    "libmuonpi-" + Version::libmuonpi::string());
 
-    /**
-     * @brief http_response
-     * Construct a default response
-     * @param req The request type
-     */
-    explicit http_response(request_type& req);
+  /**
+   * @brief http_response
+   * Construct a default response
+   * @param req The request type
+   */
+  explicit http_response(request_type &req);
 
-    /**
-     * @brief commit Complete the response
-     * @param body The body to attach to the response
-     */
-    [[nodiscard]] auto commit(std::string body) -> response_type;
+  /**
+   * @brief commit Complete the response
+   * @param body The body to attach to the response
+   */
+  [[nodiscard]] auto commit(std::string body) -> response_type;
 
-    /**
-     * @brief operator () Complete the response
-     * @param body The body to attach to the response
-     */
-    [[nodiscard]] auto operator()(std::string body) -> response_type;
+  /**
+   * @brief operator () Complete the response
+   * @param body The body to attach to the response
+   */
+  [[nodiscard]] auto operator()(std::string body) -> response_type;
 
 private:
-    response_type m_response;
+  response_type m_response;
 };
 
 template <beast::http::status Status>
-http_response<Status>::http_response(request_type &req, const content_type &content, const std::string &application_name)
-    : m_response{Status, req.version()}
-{
-    m_response.set(beast::http::field::server, application_name);
-    m_response.set(beast::http::field::content_type, content.string);
-    m_response.keep_alive(req.keep_alive());
+http_response<Status>::http_response(request_type &req,
+                                     const content_type &content,
+                                     const std::string &application_name)
+    : m_response{Status, req.version()} {
+  m_response.set(beast::http::field::server, application_name);
+  m_response.set(beast::http::field::content_type, content.string);
+  m_response.keep_alive(req.keep_alive());
 }
 
 template <beast::http::status Status>
 http_response<Status>::http_response(request_type &req)
-    : http_response<Status>{req, content_type::html()}
-{
+    : http_response<Status>{req, content_type::html()} {}
+
+template <beast::http::status Status>
+auto http_response<Status>::commit(std::string body) -> response_type {
+  m_response.body() = std::move(body);
+  m_response.prepare_payload();
+  return std::move(m_response);
 }
 
 template <beast::http::status Status>
-auto http_response<Status>::commit(std::string body) -> response_type
-{
-    m_response.body() = std::move(body);
-    m_response.prepare_payload();
-    return std::move(m_response);
-}
-
-template <beast::http::status Status>
-auto http_response<Status>::operator()(std::string body) -> response_type
-{
-    return std::move(commit(std::move(body)));
+auto http_response<Status>::operator()(std::string body) -> response_type {
+  return std::move(commit(std::move(body)));
 }
 
 } // namespace muonpi::http

--- a/include/muonpi/http_response.h
+++ b/include/muonpi/http_response.h
@@ -20,12 +20,12 @@ struct content_type {
     /**
      * @brief html Get the content type string text/html
      */
-    [[nodiscard]] static auto html() noexcept -> content_type;
+    [[nodiscard]] inline static auto html() noexcept -> content_type;
 
     /**
      * @brief json Get the content type string text/json
      */
-    [[nodiscard]] static auto json() noexcept -> content_type;
+    [[nodiscard]] inline static auto json() noexcept -> content_type;
 
     std::string string {};
 };

--- a/src/http_response.cpp
+++ b/src/http_response.cpp
@@ -3,12 +3,14 @@
 
 namespace muonpi::http {
 
-auto content_type::html() noexcept -> content_type {
-  return content_type{"text/html"};
+auto content_type::html() noexcept -> content_type
+{
+    return content_type { "text/html" };
 }
 
-auto content_type::json() noexcept -> content_type {
-  return content_type{"text/json"};
+auto content_type::json() noexcept -> content_type
+{
+    return content_type { "text/json" };
 }
 
 } // namespace muonpi::http

--- a/src/http_response.cpp
+++ b/src/http_response.cpp
@@ -13,33 +13,4 @@ auto content_type::json() noexcept -> content_type
     return content_type { "text/json" };
 }
 
-template <beast::http::status Status>
-http_response<Status>::http_response(request_type& req, const content_type& content, const std::string& application_name)
-    : m_response { Status, req.version() }
-{
-    m_response.set(beast::http::field::server, application_name);
-    m_response.set(beast::http::field::content_type, content.string);
-    m_response.keep_alive(req.keep_alive());
-}
-
-template <beast::http::status Status>
-http_response<Status>::http_response(request_type& req)
-    : http_response<Status> { req, content_type::html() }
-{
-}
-
-template <beast::http::status Status>
-auto http_response<Status>::commit(std::string body) -> response_type
-{
-    m_response.body() = std::move(body);
-    m_response.prepare_payload();
-    return std::move(m_response);
-}
-
-template <beast::http::status Status>
-auto http_response<Status>::operator()(std::string body) -> response_type
-{
-    return std::move(commit(std::move(body)));
-}
-
 } // namespace muonpi::http

--- a/src/http_response.cpp
+++ b/src/http_response.cpp
@@ -3,14 +3,12 @@
 
 namespace muonpi::http {
 
-auto content_type::html() noexcept -> content_type
-{
-    return content_type { "text/html" };
+auto content_type::html() noexcept -> content_type {
+  return content_type{"text/html"};
 }
 
-auto content_type::json() noexcept -> content_type
-{
-    return content_type { "text/json" };
+auto content_type::json() noexcept -> content_type {
+  return content_type{"text/json"};
 }
 
 } // namespace muonpi::http

--- a/src/http_response.cpp
+++ b/src/http_response.cpp
@@ -1,0 +1,45 @@
+#include "muonpi/http_response.h"
+#include "muonpi/http_tools.h"
+
+namespace muonpi::http {
+
+auto content_type::html() noexcept -> content_type
+{
+    return content_type { "text/html" };
+}
+
+auto content_type::json() noexcept -> content_type
+{
+    return content_type { "text/json" };
+}
+
+template <beast::http::status Status>
+http_response<Status>::http_response(request_type& req, const content_type& content, const std::string& application_name)
+    : m_response { Status, req.version() }
+{
+    m_response.set(beast::http::field::server, application_name);
+    m_response.set(beast::http::field::content_type, content.string);
+    m_response.keep_alive(req.keep_alive());
+}
+
+template <beast::http::status Status>
+http_response<Status>::http_response(request_type& req)
+    : http_response<Status> { req, content_type::html() }
+{
+}
+
+template <beast::http::status Status>
+auto http_response<Status>::commit(std::string body) -> response_type
+{
+    m_response.body() = std::move(body);
+    m_response.prepare_payload();
+    return std::move(m_response);
+}
+
+template <beast::http::status Status>
+auto http_response<Status>::operator()(std::string body) -> response_type
+{
+    return std::move(commit(std::move(body)));
+}
+
+} // namespace muonpi::http

